pkg/tls/tls.go

```diff
@@ -52,3 +57,59 @@ func MakeCA(signerName string) (*crypto.CA, error) {
 		Config:          caConfig,
 	}, nil
 }
+
+const (
+	TigeraCASecretName          = "tigera-ca-private"
+	TrustedCertConfigMapName    = "tigera-ca-bundle"
+	TrustedCertConfigMapKeyName = "tigera-ca-bundle.crt"
+	TrustedCertVolumeMountPath  = "/etc/pki/tls/certs/"
+	TrustedCertBundleMountPath  = "/etc/pki/tls/certs/tigera-ca-bundle.crt"
+)
+
+// TigeraCA is a signer of new certificates and has methods to retrieve existing KeyPairs and Certificates. If a user
+// brings their own secrets, TigeraCA will preserve and return them.
+type TigeraCA interface {
+	// CertificateManagement returns the CertificateManagement object or nil if it is not configured.
+	CertificateManagement() *operatorv1.CertificateManagement
+	// GetKeyPair returns an existing KeyPair. If the KeyPair is not found, a k8s.io NotFound error is returned.
+	GetKeyPair(cli client.Client, secretName, secretNamespace string) (KeyPair, error)
+	// GetOrCreateKeyPair returns a KeyPair. If one exists, some checks are performed. Otherwise, a new KeyPair is created.
+	GetOrCreateKeyPair(cli client.Client, secretName, secretNamespace string, dnsNames []string) (KeyPair, error)
+	// GetCertificate returns a Certificate. If the certificate is not found a k8s.io NotFound error is returned.
+	GetCertificate(cli client.Client, secretName, secretNamespace string) (Certificate, error)
+	// Issued returns true if the provided certificate was signed by TigeraCA and returns false for user provided KeyPairs or Certificate Management.
+	Issued(certificate Certificate) bool
+	KeyPair
+}
+
+// KeyPair wraps a Secret object that contains a private key and a certificate. Whether CertificateManagement is
+// configured or not, KeyPair returns the right InitContainer, Volumemount or Volume (when applicable).
+type KeyPair interface {
+	//UseCertificateManagement returns true if this key pair was not user provided and certificate management has been configured.
+	UseCertificateManagement() bool
+	// BYO returns true if this KeyPair was provided by the user. If BYO is true, UseCertificateManagement is false.
+	BYO() bool
```

Comment:
This might be small, but what was the reasoning behind the logic change of "check if it belongs to the operator" to "check if it belongs to the user"? I think there's a logical aspect here that we can probably only check if the operator did or didn't issue the certificate, maybe we should continue to phrase it as "IsOperatorIssued".

Reply:
There are the following scenario's:
- The secret is brought by a user
- The secret is brought by an older operator issuer prior to this pr, in which case it is not issued by the tigera ca
- The secret is created by tigera ca
- The secret is created by certificate managemed, in which case it also is not issued by the tigera ca

I started out the way you proposed, but along the way I realized the code would flow much better with this method.